### ISSUE TITLE
Improvement: stream input

### ISSUE THREAD

**Issue by ikari-pl** [jwalton/node-heapsnapshot-parser#1]
Hi! Thanks for your project, which I'm just about to use. 
From the documentation I don't see the feature that would make it a bit more awesome and maybe less memory-costly — parse stream input instead of a huge object. Assume I want to do it on the fly without storing the huge JSON anywhere. I am connected to Chrome debugger and am receiving the JSON in small chunks. Even with `oboe` library there's nothing to parse until I get 50% of it (because of the size of the arrays), but still that would be the way to move on. I hope to not store the whole JSON and its whole parsed object in memory at any point (probably impossible), instead jump directly to the much nicer objects that you provide.
Could it be possible? :)

**Comment by jwalton** [comment]
Yeah, I was thinking about this while I was writing it, actually.  I'd
welcome a PR.  :)
